packages/flutter_tools/lib/src/commands/build_ipa.dart

```diff
@@ -0,0 +1,154 @@
+        _fileSystem = fileSystem,
+        _logger = logger,
+        _processUtils =
+            ProcessUtils(processManager: processManager, logger: logger) {
+    argParser.addOption(
+      'export-options-plist',
+      valueHelp: 'ExportOptions.plist',
+      help:
+          'Path to the "-exportOptionsPlist" file passed to Xcode. See "man xcodebuild" for available keys.',
+    );
+    // Flavors may have different target names, and therefore different archive output paths.
+    usesFlavorOption();
+  }
+
+  Platform _platform;
+  XcodeProjectInterpreter _xcodeProjectInterpreter;
+  FileSystem _fileSystem;
+  Logger _logger;
+  ProcessUtils _processUtils;
+
+  @override
+  final String name = 'ipa';
+
+  @override
+  final String description =
+      'Build an iOS App Store package after running "flutter build xcarchive" (Mac OS X host only).';
+
+  String get exportOptionsPlist => stringArg('export-options-plist');
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    // Populate the "globals" that are not injected from tests.
+    _platform ??= globals.platform;
+    _xcodeProjectInterpreter ??= globals.xcodeProjectInterpreter;
+    _fileSystem ??= globals.fs;
+    _logger ??= globals.logger;
+    _processUtils ??=
+        ProcessUtils(processManager: globals.processManager, logger: _logger);
+
+    if (!_platform.isMacOS) {
+      throwToolExit('Building for iOS is only supported on macOS.');
+    }
+
+    if (exportOptionsPlist == null) {
+      throwToolExit(
+          '--export-options-plist file is required. See "man xcodebuild" for available keys.');
+    }
+
+    final FileSystemEntityType type = _fileSystem.typeSync(exportOptionsPlist);
+    if (type == FileSystemEntityType.notFound) {
+      throwToolExit(
+          '"$exportOptionsPlist" property list does not exist. See "man xcodebuild" for available keys.');
+    } else if (type != FileSystemEntityType.file) {
+      throwToolExit(
+          '"$exportOptionsPlist" is not a file. See "man xcodebuild" for available keys.');
+    }
+
+    if (!_xcodeProjectInterpreter.isInstalled) {
+      throwToolExit('Cannot find "xcodebuild". Run "flutter doctor".');
+    }
+
+    final BuildInfo buildInfo =
+        getBuildInfo(forcedBuildMode: BuildMode.release);
+
+    final BuildableIOSApp app = await applicationPackages.getPackageForPlatform(
+      TargetPlatform.ios,
+      buildInfo,
+    ) as BuildableIOSApp;
+
+    if (app == null) {
+      throwToolExit('Application not configured for iOS.');
+    }
+
+    final String archiveBundleOutputPath =
+        _fileSystem.path.absolute(app.archiveBundleOutputPath);
+    if (!_fileSystem.directory(archiveBundleOutputPath).existsSync()) {
+      final String flavorFlag =
+          buildInfo.flavor == null ? '' : ' --flavor ${buildInfo.flavor}';
+      throwToolExit('xcarchive not found at $archiveBundleOutputPath. '
+          'Run "flutter build xcarchive$flavorFlag" to generate it.');
+    }
```

Comment:
if we do wrap the xcarchive part, it would be nice to have some descriptions around what the flavor translates to in the UI if the asked flavor isn't present.